Added support to upload images

Jamezz opened the pull request:
A little hacky way to upload images, but it works decently in my environment. I ended up intercepting any single line messages hubot that contains a image file extension, downloading the image to hubot, uploading it via tg and then deleting the image in the temp directory...

Commit:
Add support to upload images

Watch the string flow and intercept any messages containing .jpg or
.png. Download the images to temp and send it via tg.

Jamezz committed:
Pull the image temp dir out to env variable

Defaults to the precoded path in previous commit, but the
HUBOT_TG_TMPDIR variable will override.

zenitraM commented on src/tg.coffee:
```diff
@@ -1,13 +1,27 @@
     @port = process.env['HUBOT_TG_PORT'] || 1123
     @host = process.env['HUBOT_TG_HOST'] || 'localhost'
+    @imageExtensions = [".jpg",".png", ".jpeg"]
+    @tempdir = process.env['HUBOT_TG_TMPDIR'] || '/srv/hubot/bin/downloads/'
```
Hey, could you change this tempdir to something else in https://nodejs.org/api/os.html#os_os_tmpdir?

Jamezz committed:
Change tmp and stop deleting files so fast

Switch to node's definition of a temporary directory so we don't have to
worry about it.

Also stop deleting images as the message gets sent to telegram as this
can cause a race condition on slow download and cause the image to never
be sent. It's not perfect but it works.
